src/main/groovy/com/bmuschko/gradle/java2html/GenerateOverviewTask.groovy

```diff
@@ -26,14 +26,23 @@ import org.gradle.api.tasks.*
  *
  * @author Benjamin Muschko
  */
+@CacheableTask
 class GenerateOverviewTask extends DefaultTask {
+    @PathSensitive(PathSensitivity.RELATIVE)
     @InputFiles FileCollection srcDirs
+
+    @PathSensitive(PathSensitivity.RELATIVE)
     @OutputDirectory File destDir
-    @Input String pattern
-    @Input String windowTitle
-    @Input String docTitle
-    @Input String docDescription
+
+    @Input @Optional String pattern
```

Comment:
I had a look at the plugin code again and the task. At the moment we don't have any handling in place that would _not_ pass the attributes to Ant if the value is null. Not sure how the Ant task would behave. Let's remove `@Optional` to ensure that a value is being enforced.

Reply:
OK